src/pytorch_lightning/strategies/ipu.py

```diff
@@ -220,15 +231,20 @@ def inference_opts(self) -> "poptorch.Options":
             self._inference_opts = self._create_opts(training=False)
         return self._inference_opts
 
+    @property
+    def lightning_module(self) -> Optional["pl.LightningModule"]:
+        if self.model is not None:
+            return unwrap_lightning_module(self.model)
+
     def _convert_to_poptorch_loader(
-        self, dataloader: DataLoader, sampler, mode: Optional[RunningStage] = None
+        self, dataloader: DataLoader, sampler: Union[Sampler, Iterable], mode: Optional[RunningStage] = None
     ) -> "poptorch.DataLoader":
         if isinstance(dataloader, poptorch.DataLoader):
             # the user is returning the `poptorch.DataLoader` directly, don't change anything.
             return dataloader
 
         dl_args, dl_kwargs = _get_dataloader_init_args_and_kwargs(
-            dataloader, sampler, mode, self.replication_factor > 1
+            dataloader, sampler, mode, self.replication_factor > 1  # type:ignore [arg-type]
```

Comment:
Let's list the args vertically, so it becomes clear which argument we are ignoring:
```suggestion
            dataloader, 
            sampler,  # type: ignore[arg-type]
            mode, 
            self.replication_factor > 1
```